crates/ruff/src/rules/flake8_raise/rules/unnecessary_paren_on_raise_exception.rs

```diff
@@ -52,6 +52,18 @@ pub(crate) fn unnecessary_paren_on_raise_exception(checker: &mut Checker, expr:
         range: _,
     }) = expr
     {
+        let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+            return;
+        };
+
+        let scope = checker.semantic().scope();
+        if let Some(binding_id) = scope.get(id) {
+            let binding = checker.semantic().binding(binding_id);
+            if binding.kind.is_function_definition() {
+                return;
+            }
+        }
```

Comment:
I'm not opposed to adding this, since it seems like an improvement over what we're doing now, but it's worth noting that it will miss a lot of cases, including the case from the linked issue, where they had a method within a class:

```py
class CustomError(Exception):
    def __init__(self, msg: str):
        self.msg = msg

    @staticmethod
    def timeout():
        return CustomError("Operation timed out!")

raise CustomError.timeout()
```

Reply:
I might have misunderstood, but wasn't the point of the issue that we wanted to not flag the above snippet? Or do we want to flag it as a violation but just not autofix it? If the latter, do we have precedent for conditional fixing? Or would it be better to just keep flagging broadly but make the autofix `suggested` or `manual`?

Reply:
That's right -- we _don't_ want to flag the above. But, I think the fix you have here would work for the wrong reason, which is that it's now avoiding flagging attribute accesses (`a.b()`).

Reply:
Clear, thanks for clarifying. Been trying to implement logic, which should only flag class method calls and not other attributes with the following:

```rust
match func.as_ref() {
    // Exclude function calls
    Expr::Name(ast::ExprName { id, .. }) => {
        let scope = checker.semantic().scope();
        if id_is_function_definition(checker, scope, id) {
            return;
        }
    }
    // Exclude Class method calls
    Expr::Attribute(ast::ExprAttribute { attr, value, .. }) => {
        if let Expr::Name(ast::ExprName { id, .. }) = value.as_ref() {
            if let Some((scope_id, _)) = checker.semantic().nonlocal(id) {
                let scope = &checker.semantic().scopes[scope_id];
                if id_is_function_definition(checker, scope, attr.as_str()) {
                    return;
                }
            }
        }
    }
    _ => (),
};

...

fn id_is_function_definition(checker: &Checker, scope: &Scope, id: &str) -> bool {
    if let Some(binding_id) = scope.get(id) {
        let binding = checker.semantic().binding(binding_id);
        return binding.kind.is_function_definition();
    }
    false
}
```

But that's still flagging the scenario where: 

```python
class CustomError(Exception):
    def __init__(self, msg: str):
        self.msg = msg

    @staticmethod
    def timeout():
        return CustomError("Operation timed out!")

raise CustomError.timeout()
```

I'm guessing my estimation of how the scopes work is off. Any input on the most efficient/clear way to check whether the `Expr::Attribute` relates to a method call or a different attribute?

Reply:
Blocked by https://github.com/astral-sh/ruff/issues/5487